kotlinjs/src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,111 @@
+import kotlinx.browser.document
+import kotlinx.browser.window
+import org.w3c.dom.HTMLDivElement
+import org.w3c.dom.HTMLHeadingElement
+import kotlin.js.Date
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.floor
+import kotlin.math.sin
+
+fun main() {
+    val clockFace = document.getElementById("clock-face") as HTMLDivElement
+    val clockTime = document.getElementById("clock-time") as HTMLHeadingElement
+
+    // prepare clock digit
+    IntRange(0, 9)
+        .map { toDigitPosition(it.toDouble()) }
+        .map { toDigitElement(it) }
+        .forEach { clockFace.appendChild(it) }
+
+    // prepare clock tick
+    IntRange(0, 99)
+        .map { toTickProperty(it) }
+        .map { toTickElement(it) }
+        .forEach { clockFace.appendChild(it) }
+
+    window.requestAnimationFrame { render(clockFace, clockTime) }
+}
+
+fun render(clockFace: HTMLDivElement, clockTime: HTMLHeadingElement) {
+    val date = Date()
+    val timeOfDay = (date.getHours() * 60 * 60 * 1000) +
+            (date.getMinutes() * 60 * 1000) +
+            (date.getSeconds() * 1000) +
+            date.getMilliseconds()
+    val fractionOfDay = timeOfDay / 86400000.0
+
+    renderFace(clockFace, fractionOfDay)
+    renderTime(clockTime, fractionOfDay)
+
+    window.requestAnimationFrame { render(clockFace, clockTime) }
+}
+
+fun toDigitPosition(i: Double): Triple<Double, Double, Double> {
```

Comment:
```suggestion
fun Double.toDigitPosition(): Triple<Double, Double, Double> {
```

pake extension biar idiomatic. jadi nanti usagenya:

```kt
IntRange(0, 9)
    .map { it.toDouble().toDigitPosition() }
```

Reply:
`Triple<Double, Double, Double>` juga bisa dibikinin type biar strict:

```kt
data class DigitPosition(val i: Int, val x: Double, y: Double)
```

jadi nanti methodnya bisa pake expression body:

```kt
fun Double.toDigitPosition() = DigitPosition(
    i = this,
    x = 50 - sin((PI * this) / 5) * 40,
    y = 50 + cos((PI * this) / 5) * 40
)
```

Reply:
Kalau mau direfactor lebih jauh lagi, kita bisa pakai `Int` sebagai receiver type. jadi pemanggilannya ngga perlu `.toDouble()` dulu